### PR TITLE
fix: Always Send `Content-Type: application/json`

### PR DESCRIPTION
**Pull request by sleepdefic1t**
## Proposed changes

Per "Always send `Content-Type: application/json`" #20 

Set header `content-type` for IoT and OS builds.

IoT: 
>` http/iot/http.cpp`
```diff
- http.addHeader("Content-Type", "application/x-www-form-urlencoded");
+ http.addHeader("Content-Type", "application/json");
```

OS:
> `http/os/http.cpp`
```diff
+ struct curl_slist *header = NULL;
+ curl_slist_append(header, "Content-Type: application/json");
+ curl_easy_setopt(curl, CURLOPT_HTTPHEADER, header);
```

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

Test fail due to missing ms in timestamp strings, #22 resolves this.

### CODE REVIEW

File: src/http/os/http.cpp [hunk 1]
```diff
@@ -73,7 +73,14 @@ class PlatformHTTP : public AbstractHTTP
...
                 curl_easy_setopt(curl, CURLOPT_URL, ss.str().c_str()); // Set the URL that is about to receive our POST
-                curl_easy_setopt(curl, CURLOPT_POSTFIELDS, body); // Now specify the POST data ex: "username=baldninja"
+
+                /* make header with content-type */
+                struct curl_slist *header = NULL;
+                curl_slist_append(header, "Content-Type: application/json");
+
+                /* set header & body */
+                curl_easy_setopt(curl, CURLOPT_HTTPHEADER, header); // Set Header
+                curl_easy_setopt(curl, CURLOPT_POSTFIELDS, body); // Now specify the POST json data ex: "username=baldninja"
```
Comment (ciband) [inline src/http/os/http.cpp]: typo in "boldninja", unless Rok is having hair issues, in which case my apologies for being insensitive.

**Comment by sleepdefic1t** [comment]
No typo 😂

https://dexplorer.ark.io:8443/api/v2/delegates/baldninja

File: src/http/os/http.cpp (hunk 1 above)
Comment (sleepdefic1t) [reply src/http/os/http.cpp]: No typo 😂

https://dexplorer.ark.io:8443/api/v2/delegates/baldninja

**Comment by sleepdefic1t** [comment]
Test fail due to #22.

This should be merged.

**Comment by faustbrian** [comment]
This PR was automatically closed because you referenced it with a keyword.

**Pull request closed without merging**